### PR TITLE
enhance(blobnode):  optimize data inspect, skip init broken disk

### LINKED ISSUE #2767: [Enhancement]: If a node has a bad disk, the blobnode service will not start by default config
### Contact Details

_No response_

### Is there an existing issue for this?

- [X] I have searched all the existing issues

### Priority

low (Default)

### Environment

```markdown
- CubeFS version: blobstore-v1.2.3
- Deployment mode(docker or standalone or cluster):
- Dependent components:
- OS kernel version(Ubuntu or CentOS):
- CPU/Memory:
- Others:
```

### Current Behavior

use default config, start blobnode service, when the node has a bad disk

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### CubeFS Log

_No response_

### Anything else? (Additional Context)

_No response_

### LINKED ISSUE #2768: [Feature]: support hot-set the switch/rate without restarting the blobnode service
### Contact Details

_No response_

### Is there an existing issue for this?

- [X] I have searched all the existing issues

### Is your feature request related to a problem? Please describe.

Now, in the blobnode service, every time the check switch is turned on, the blobnode service needs to be restarted, which takes a lot of time.
So:  
 1. We need a way to hot-set the switch without restarting the service
 2. We need to set the inspection rate

It will be committed to the branch blobstore-v1.2.4

### Describe the solution you'd like.

_No response_

### Describe an alternate solution.

_No response_

### Anything else? (Additional Context)

_No response_

### PR DESCRIPTION
**Pull request by mawei029**
**What this PR does / why we need it**:
1. support the inpect switch can be set from clustermgr(unified interface), set inspect rate limit from http
2. skip init new disk storage, when the disk is broken

**Which issue this PR fixes**:

fixes #2767  
fixes #2768

**Special notes for your reviewer**:
Pass them all, and I'll merge the commit id

**Release note**:

```release-note
```

### CODE REVIEW

**Review by JasonHu520 — APPROVED** [review]

File: blobstore/blobnode/datainspect.go [hunk 1]
```diff
@@ -2,117 +2,222 @@ package blobnode
...
-						}
-					}
-				}(ds)
+		case <-t.C:
+			if !mgr.getSwitch() {
+				continue
 			}
-			wg.Wait()
-			timer.Reset(time.Duration(s.Conf.ChunkInspectIntervalSec) * time.Second)
+			mgr.inspectAllDisks(ctx)
```
Comment (cessory) [inline blobstore/blobnode/datainspect.go:88]: 这个函数执行之后，需要Reset一下定时器

File: blobstore/blobnode/datainspect.go [hunk 2]
```diff
@@ -2,117 +2,222 @@ package blobnode
...
+	span := trace.SpanFromContextSafe(ctx)
+	t := time.NewTimer(time.Duration(mgr.conf.IntervalSec) * time.Second) // wait next round inspect
+	defer t.Stop()
+
+	disks := mgr.svr.copyDiskStorages(ctx)
+	for _, ds := range disks {
+		mgr.setLimiter(ds)
+	}
+
 	for {
```
Comment (cessory) [inline blobstore/blobnode/datainspect.go:107]: 这个for循环可以去掉

File: blobstore/blobnode/datainspect.go [hunk 3]
```diff
@@ -2,117 +2,222 @@ package blobnode
...
+						tokenSz = int(remain)
+					}
+					lmt.WaitN(ctx, tokenSz)
+					remain -= int64(tokenSz)
+				}
+
+				_, err = cs.Read(ctx, shardReader)
+				if err == bloberr.ErrOverload {
+					successCnt = 0
+					mgr.setRate(lmt, lmt.Limit()/slowDownRatio) // slow down
```
Comment (cessory) [inline blobstore/blobnode/datainspect.go:206]: 这里不用逐步减少，直接减少到最小值，快速降低，慢慢恢复

File: blobstore/blobnode/datainspect.go [hunk 4]
```diff
@@ -2,117 +2,222 @@ package blobnode
...
+					continue
+				}
+
+				if successCnt > 0 && successCnt%speedUpCnt == 0 {
```
Comment (cessory) [inline blobstore/blobnode/datainspect.go:210]: successCnt需要++

File: blobstore/blobnode/datainspect.go [hunk 5]
```diff
@@ -146,6 +274,69 @@ func reportBadShard(cs core.ChunkAPI, blobID proto.BlobID, err error) {
...
+
+	ctx := c.Request.Context()
+	span := trace.SpanFromContextSafe(ctx)
+
+	if args.Rate < minRateLimit {
+		c.RespondError(errors.New("rate value is too small"))
+		return
+	}
+
+	if args.Rate > maxRateLimit {
```
Comment (cessory) [inline blobstore/blobnode/datainspect.go:320]: 这里不用判断maxRateLimit, 但是需要更改maxRateLimit的值

**Review by cessory — APPROVED** [review]

**Pull request merged**